tests/functional/test_neptune.py

```diff
@@ -20,8 +20,8 @@ def setUp(self):
         self.http_stubber = ClientHTTPStubber(self.client)
 
     def assert_presigned_url_injected_in_request(self, body):
-        self.assertIn('PreSignedUrl', body)
-        self.assertNotIn('SourceRegion', body)
+        self.assertIn(b'PreSignedUrl', body)
+        self.assertNotIn(b'SourceRegion', body)
```

Comment:
What's causing these changes? Were these `body`s not also `bytes` before?

Reply:
Yeah the request's `body` value was a string before. Doing some investigation for why that was the case now.

Reply:
This is the relevant bit of code: https://github.com/boto/botocore/blob/develop/botocore/awsrequest.py#L418-L420

It url encodes the body dictionary to a string, but doesn't then subsequently encode the string to bytes. It seems as though for `query` requests that gets handed off to urllib3.

Reply:
I'm not sure I understand this in response to Jonas' question. We've changed the type of the body for these unrelated tests because of a new change we added. That seems to be because we're now forcing these to bytes prematurely when we're encoding everything in the [new query transform](https://github.com/boto/botocore/pull/2959/files#diff-d5dc8bb337890042943977b62e8bb2e7db71415d35bb17a41ad4bf4780fb3617R3435-R3445). That's done indiscriminately regardless of compression.

Do we have a reason we need to do that there? This feels like a very likely case we break something.

Reply:
Yeah, it looks like we broke most of these tests and then updated them to match the breakage. Removing [this line](https://github.com/boto/botocore/pull/2959/files#diff-d5dc8bb337890042943977b62e8bb2e7db71415d35bb17a41ad4bf4780fb3617R3445) that was added avoids this issue and doesn't seem to affect any of the tests.

Can we take another look at this on Monday? If there's not something I'm missing let's revert it.

Reply:
Thanks for catching this!

I wouldn't say the tests are unrelated necessarily. There is a section in the design doc that discusses the handlers that generate the presigned URLs which get injected into the bodies of some operations, and these are the ones in question. 

However, I don't see any harm in delaying encoding to bytes until later in the operation call if we're worried about potential breakages. I'll take another look Monday, but it should be fine to revert it.